code/modules/power/supermatter/supermatter_hit_procs.dm

```diff
@@ -274,12 +276,14 @@
 				message_admins("[src] has consumed [consumed_object], [suspicion] [ADMIN_JMP(src)].")
 			investigate_log("has consumed [consumed_object] - [suspicion].", INVESTIGATE_ENGINE)
 		qdel(consumed_object)
-	if(!iseffect(consumed_object) && power_changes)
-		matter_power += 200
+	if(!iseffect(consumed_object) && isitem(consumed_object) && power_changes)
+		var/obj/item/consumed_item = consumed_object
+		object_size = consumed_item.w_class
+		matter_power += 70 * object_size
 
 	//Some poor sod got eaten, go ahead and irradiate people nearby.
-	radiation_pulse(src, max_range = 6, threshold = 0.3, chance = 30)
-	for(var/mob/living/near_mob in range(10))
+	radiation_pulse(src, max_range = 2 * object_size, threshold = 0.6 / object_size, chance = 10 * object_size)
+	for(var/mob/living/near_mob in range(2 * object_size))
```

Comment:
I'd prefer to lean on the side of too little for these, I don't trust radiation code to behave sanely. I'd prefer if you treated the old numbers as if they were using 4 rather then 3, and treated the max range as 5.

Reply:
i'll do the pulse like this `radiation_pulse(src, max_range = 1.5 * object_size, threshold = 1.2 / object_size, chance = 10 * object_size)` 
this way the max range has been lowered a bit to match the max_range of 5, the threshold is higher so that it's less likely to pass with smaller objects and the chance is the same.
for the for() loop i'll keep the 2 so that is slightly bigger than the actual pulse so you know that something's going on before going near the danger zone